tests/test_sorting.py

```diff
@@ -0,0 +1,367 @@
+if len(EXPTIME_EMGAIN_SEC) != len(EM_EMGAIN):
+    raise Exception(f'Inconsistent number of sets in EM-gain vs DAC')
+if len(EXPTIME_EMGAIN_SEC) != len(NFRAMES_EMGAIN):
+    raise Exception(f'Inconsistent number of sets in EM-gain vs DAC')
+# Values
+cmdgain_emgain, exptime_emgain = get_cmdgain_exptime_emgain(
+    em_emgain = EM_EMGAIN,
+    exptime_emgain_sec = EXPTIME_EMGAIN_SEC,
+    nframes = NFRAMES_EMGAIN,
+    )
+if len(cmdgain_emgain) != len(exptime_emgain):
+    raise Exception(f'Inconsistent lengths in em-gain vs dac')
+# Total number of frames
+n_emgain_total = len(cmdgain_emgain)
+
+# DRP Dataset
+# Create directory for temporary data files (not tracked by git)
+if not os.path.exists(Path('simdata')):
+    os.mkdir(Path('simdata'))
+
+# Loop over the two cases of non-linearity
+change_exptime = [False, True]
+idx_frame = 0
+for change in change_exptime:
+    filename_list = []
+    if change == False:
+        cmdgain_nonlin = cmdgain_nonlin_wo_change
+        exptime_nonlin = exptime_nonlin_wo_change
+    elif change == True:
+        cmdgain_nonlin = cmdgain_nonlin_w_change
+        exptime_nonlin = exptime_nonlin_w_change
+    else:
+        raise Exception('Undefined choice for non-linearity')
+    n_nonlin_total = len(exptime_nonlin)
+    # Mean frame
+    print('Generating frames for mean frame')
+    for i_f in range(n_mean_frame_total):
+        filename = make_minimal_image(
+            cmdgain=cmdgain_mean_frame[i_f],
+            exptime_sec=exptime_mean_frame[i_f],
+            frameid=idx_frame,
+            )
+        filename_list += [filename]
+        idx_frame += 1
+    # K-gain
+    print('Generating frames for k-gain')
+    for i_f in range(n_kgain_total):
+        filename = make_minimal_image(
+            cmdgain=cmdgain_kgain[i_f],
+            exptime_sec=exptime_kgain[i_f],
+            frameid=idx_frame,
+            )
+        filename_list += [filename]
+        idx_frame += 1
+    # Non-linearity
+    print('Generating frames for non-linearity')
+    for i_f in range(n_nonlin_total):
+        filename = make_minimal_image(
+            cmdgain=cmdgain_nonlin[i_f],
+            exptime_sec=exptime_nonlin[i_f],
+            frameid=idx_frame,
+            )
+        filename_list += [filename]
+        idx_frame += 1
+    # EM-gain
+    print('Generating frames for em-gain')
+    for i_f in range( n_emgain_total):
+        filename = make_minimal_image(
+            cmdgain=cmdgain_emgain[i_f],
+            exptime_sec=exptime_emgain[i_f],
+            frameid=idx_frame,
+            )
+        filename_list += [filename]
+        idx_frame += 1
+
+    # Shuffle file order randomnly
+    random.shuffle(filename_list)
+
+    # Create Dataset
+    dataset_cal = data.Dataset(filename_list)
+
+    # Apply sorting algorithm and check results (maybe output of sorting is
+    # mean frame and the type used as input? Instead of them all. Then, check
```

Comment:
the comment says we should check the properties, but there are no checks in this unit test. The unit test should have some asserts at the end that checks the outputs are what are expected.

Reply:
I submitted the PR as soon as I could for your review. Will do.

Reply:
Added a full set of tests consistent with the aim this PR for DRP v1.1